File: Day 27/1854.java
class Solution {
    public int maximumPopulation(int[][] logs) {

        int maxPopulation = 0;
        int answerYear = 0;

        for (int year = 1950; year <= 2050; year++) {

            int population = 0;

            for (int i = 0; i < logs.length; i++) {

                if (logs[i][0] <= year && year < logs[i][1]) {
                    population++;
                }
            }

            if (population > maxPopulation) {
                maxPopulation = population;
                answerYear = year;
            }
        }

        return answerYear;
    }
}

File: Day 27/1854.py
class Solution:
    def maximumPopulation(self, logs: list[list[int]]) -> int:
        year_changes = [0] * 101

        for birth, death in logs:
            year_changes[birth - 1950] += 1
            year_changes[death - 1950] -= 1

        max_pop = 0
        cur_pop = 0
        earliest_year = 1950

        for i in range(101):
            cur_pop += year_changes[i]
            if cur_pop > max_pop:
                max_pop = cur_pop
                earliest_year = 1950 + i

        return earliest_year
